fendermint/ipc/src/message.rs

```diff
@@ -0,0 +1,16 @@
+//! IPC messages
+
+use crate::pof::IPCParentFinality;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum IPCMessage {
+    TopDown(IPCParentFinality),
+    BottomUp,
+}
+
+impl From<IPCMessage> for Vec<u8> {
+    fn from(value: IPCMessage) -> Self {
+        serde_json::to_vec(&value).expect("should not happen")
```

Comment:
Ah, this is super confusing actually. I read it as _parsing_ `Vec<u8>` like 3 times already. 

So yes, this will (probably) not fail because we can always convert to JSON, and this isn't the case when we're running the risk of a malicious validator passing us some untrustworthy byte content. However, I'd argue this implementation should not exist: it is not an inherent property of `IPCMessage` that it's transformed to binary via JSON. Case in point: we actually use IPLD. Let's not decide for everyone what they should use, and leave it up to the application implementer.